src/views/ProjectView.vue

```diff
@@ -1,9 +1,127 @@
+  try {
+    const selectedImages = store.state.selectedImages
+    const inputData = selectedImages.map(image => ({
+      'source': image.image,
+      'basename': image.basefile_name
+    }))
+
+    const requestBody = {
+      'name': session.name,
+      'input_data': inputData
+    }
+
+    const url = 'http://127.0.0.1:8000/api/datasessions/' + session.id + '/'
+    const data = await fetchApiCall(url, 'PATCH', requestBody)
+
+  } catch (error) {
+    console.error('Error importing images:', error)
+  }
+}
+
+const selectDataSession = (session) => {
+  isPopupVisible.value = false
+  addImagesToExistingSession(session)
+  router.push({ name: 'DataSessions' })
+}
+
+const createNewDataSession = async () => {
+    if (doesSessionNameExist(newSessionName.value)) {
+        errorMessage.value = 'Data Session name already exists. Please choose a different name.'
+        return
+    }
+
+    try {
+        const selectedImages = store.state.selectedImages
+        const inputData = selectedImages.map(image => ({
+            'source': image.image,
+            'basename': image.basefile_name
+        }))
+        const requestBody = { 
+            'name': newSessionName.value,
+            'input_data': inputData 
+        }
+        const url = 'http://127.0.0.1:8000/api/datasessions/'
+        const data = await fetchApiCall(url, 'POST', requestBody)
+
+        isPopupVisible.value = false
+        newSessionName.value = ''
+        errorMessage.value = ''
+
+        router.push({ name: 'DataSessions' })
+    } catch (error) {
+        console.log('Error creating new data session:', error)
+        errorMessage.value = 'Error creating new data session'
+    }
+}
+
+const doesSessionNameExist = (name) => {
+    return uniqueDataSessions.value.some(session => session.name === name)
+}
+
 </script>
 
 <template>
     <ToggleButton/>
-    <ImportButton/>
-</template>
+    <v-btn @click="getDataSessions">Add to a Session</v-btn>
+    
+    <v-dialog v-model="isPopupVisible" width="300">
+        <v-card>
+            <v-card-title>Data Sessions</v-card-title>
+            <v-card-text>
+            <v-list>
+                <v-list-item v-for="session in uniqueDataSessions" :key="session.id" @click="selectDataSession(session)">
+                <v-list-item-content>
+                    {{ session.name }}
+                </v-list-item-content>
+                </v-list-item>
+            </v-list>
+            <!-- Input for new session name -->
+            <v-text-field v-model="newSessionName" label="New Session Name" />
+            <!-- Error message -->
+            <div v-if="errorMessage">{{ errorMessage }}</div>
+            </v-card-text>
+            <v-card-actions>
+            <v-btn color="primary" text @click="createNewDataSession">Create New Session</v-btn>
+            <v-btn color="primary" text @click="isPopupVisible = false">Close</v-btn>
+            </v-card-actions>
+        </v-card>
+    </v-dialog>
+  </template>
+  
```

Comment:
Missing newline.

Reply:
isn't line 127 the newline?

Reply:
am I misunderstanding something?

Reply:
You can read here - more specifically, GitHub is showing the file isn't properly terminated. https://www.baeldung.com/linux/files-end-with-newlines

Reply:
<img width="1728" alt="Screenshot 2024-01-11 at 9 32 31 AM" src="https://github.com/LCOGT/datalab-ui/assets/54489472/756ece89-b7d2-444d-b3a6-2c8d10f64d20">
<img width="568" alt="Screenshot 2024-01-11 at 9 32 55 AM" src="https://github.com/LCOGT/datalab-ui/assets/54489472/ba45dfc9-8608-44d4-9bef-9489a76b8926">

Reply:
github is gaslighting